LukeberryPi/blog
# Add RSS Feed Generation Script

**Description:**
This PR introduces a script to generate an RSS feed for the blog. The script reads the markdown files from the `src/articles` directory, extracts metadata using `gray-matter`, and creates an RSS feed using the `rss` library. The feed is then written to the `public/rss.xml` file.

**Changes Made:**
- Added `dotenv` to handle environment variables.
- Created `gen-rss.js` script to generate the RSS feed.
- Updated the `package.json` with a new script command: `generate:rss`.
- Configured the script to use `SITE_URL` from environment variables for generating URLs.

**How to Test:**
1. Ensure you have a `.env.local` file in the project root with the following content:
   ```
   SITE_URL=https://lukeberrypi.vercel.app/
   ```
2. Run the following command to generate the RSS feed:
   ```bash
   npm run generate:rss
   ```
3. Verify that `public/rss.xml` is generated and contains the correct URLs and post data.

**Notes:**
- The script is designed to be run locally and in production, using environment variables to configure the site URL.
- In development, it defaults to `http://localhost:3000` if `SITE_URL` is not set.
